imap_processing/swapi/tests/test_swapi_l1.py

```diff
@@ -1 +1,344 @@
+    # TODO: update test when we update MODE from HVENG to HVSCI
+    ds["MODE"] = xr.DataArray(np.repeat(np.arange(total_sweeps), 12))
+    bad_data_indices = check_for_bad_data(ds)
+    np.testing.assert_array_equal(bad_data_indices, np.arange(0, 24))
+
+    # Check for bad SWEEP_TABLE data.
+    # Reset MODE data and create first sweep to be mixed value
+    ds["MODE"] = xr.DataArray(np.full((total_sweeps * 12), 2))
+    ds["SWEEP_TABLE"][:12] = np.arange(0, 12)
+    np.testing.assert_array_equal(check_for_bad_data(ds), np.arange(0, 12))
+
+    ds["SWEEP_TABLE"][24:] = np.arange(0, 12)
+    expected = [
+        0,
+        1,
+        2,
+        3,
+        4,
+        5,
+        6,
+        7,
+        8,
+        9,
+        10,
+        11,
+        24,
+        25,
+        26,
+        27,
+        28,
+        29,
+        30,
+        31,
+        32,
+        33,
+        34,
+        35,
+    ]
+    np.testing.assert_array_equal(check_for_bad_data(ds), expected)
+
+    # Check for bad PLAN_ID_SCIENCE data.
+    ds["SWEEP_TABLE"] = xr.DataArray(np.repeat(np.arange(total_sweeps), 12))
+    ds["PLAN_ID_SCIENCE"][24 : total_sweeps * 12] = np.arange(0, 12)
+    np.testing.assert_array_equal(check_for_bad_data(ds), np.arange(24, 36))
+
+
+def test_decompress_count():
+    """Test for decompress count"""
+    # Test compressed and not overflow case
+    raw_value = np.array([[12]])
+    compression_flag = np.array([[1]])
+    expected_value = raw_value[0][0] * 16
+    returned_value = decompress_count(raw_value, compression_flag)
+    assert returned_value[0][0] == expected_value
+
+    # Test compressed and overflow case
+    raw_value = np.array([[0xFFFF]])
+    compression_flag = np.array([[1]])
+    expected_value = -1
+    returned_value = decompress_count(raw_value, compression_flag)
+    assert returned_value[0][0] == expected_value
+
+    # Test not compressed case
+    raw_value = np.array([[12]])
+    compression_flag = np.array([[0]])
+    expected_value = 12
+    returned_value = decompress_count(raw_value, compression_flag)
+    assert returned_value[0][0] == expected_value
+
+    # compressed + no-overflow, compressed + overflow, no compression
+    raw_values = np.array([[12, 0xFFFF, 12]])
+    compression_flag = np.array([[1, 1, 0]])
+    expected = np.array([[12 * 16, -1, 12]])
+    returned_value = decompress_count(raw_values, compression_flag)
+    np.testing.assert_array_equal(returned_value, expected)
+
+
+def test_find_sweep_starts():
+    """Test for find sweep starts"""
+    time = np.arange(26)
+    sequence_number = time % 12
+    ds = xr.Dataset({"SEQ_NUMBER": sequence_number}, coords={"Epoch": time})
+
+    start_indices = find_sweep_starts(ds)
+    np.testing.assert_array_equal(start_indices, [0, 12])
+
+    ds["SEQ_NUMBER"].data[:12] = np.arange(3, 15)
+    start_indices = find_sweep_starts(ds)
+    np.testing.assert_array_equal(start_indices, [12])
+
+    ds["SEQ_NUMBER"] = np.arange(3, 29)
```

Comment:
Can you add a quick note about what this is testing? Just glancing at it I would have expected it to identify sequence starts...